contracts/Staking.sol

```diff
@@ -67,26 +85,27 @@ contract Staking {
 
     // Private functions
     function _stake() private {
+        require(
+            _validators.length < _maximumNumValidators,
+            "Validator set has reached full capacity"
+        );
         _stakedAmount += msg.value;
         _addressToStakedAmount[msg.sender] += msg.value;
 
         if (
             !_addressToIsValidator[msg.sender] &&
             _addressToStakedAmount[msg.sender] >= VALIDATOR_THRESHOLD
         ) {
-            // append to validator set
-            _addressToIsValidator[msg.sender] = true;
-            _addressToValidatorIndex[msg.sender] = _validators.length;
-            _validators.push(msg.sender);
+            _appendToValidatorSet(msg.sender);
         }
 
         emit Staked(msg.sender, msg.value);
     }
 
     function _unstake() private {
         require(
-            _validators.length > MINIMUM_REQUIRED_NUM_VALIDATORS,
-            "Validators can't be less than MINIMUM_REQUIRED_NUM_VALIDATORS"
+            _validators.length > _minimumNumValidators,
+            "Validators can't be less than MinimumRequiredNumValidators"
```

Comment:
Please fix this error message

Reply:
Fixed 115588a7e5c6265e64aa2ce6755bf45a100913b9